catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.suggestion;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.codice.ddf.spatial.geocoding.Suggestion;
+import org.codice.usng4j.CoordinateSystemTranslator;
+import org.codice.usng4j.DecimalDegreesCoordinate;
+import org.codice.usng4j.UsngCoordinate;
+import org.codice.usng4j.UtmUpsCoordinate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A regex-based processor for identifying potential UTM/UPS coordinate literals in a given input
+ * string despite deviations from the official spec. The general policy is to be as forgiving as
+ * possible without sacrificing determinism.
+ */
+public class UtmUpsCoordinateProcessor {
+  private static final Logger LOGGER = LoggerFactory.getLogger(UtmUpsCoordinateProcessor.class);
+
+  private static final Character WHITESPACE_CHAR = ' ';
```

Comment:
:pencil2: I don't see the advantage of this constant. There are multiple kinds of whitespace chars (`\n`, `\t`, `\r`, `\v`, etc.), making this more ambiguous by name, and `WHITESPACE_CHAR` is substantially longer than simply using a "`' '`" or "` `".

Reply:
I think it makes the usage a little more clear but I'll rename it to something less ambiguous.